ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore.hs

```diff
@@ -152,11 +154,19 @@ bracketTestKeystore withKeystore =
 -- directory.
 newTestKeystore :: IO Keystore
 newTestKeystore = liftIO $ runIdentityT $ fromKeystore $ do
-    tempDir         <- liftIO getTemporaryDirectory
-    (tempFile, hdl) <- liftIO $ openTempFile tempDir "keystore.key"
-    liftIO $ hClose hdl
-    us <- peekUserSecret tempFile
+    fp <- liftIO $ getKeystorePath
+    us <- peekUserSecret fp
     Keystore <$> newMVar (InternalStorage us)
+  where
+    getKeystorePath :: IO FilePath
+    getKeystorePath = do
+        tempDir   <- getTemporaryDirectory
+        let dir = tempDir </> "ariadne-test"
+        createDirectoryIfMissing False dir
+        (fp, hdl) <- openTempFile dir "keystore.key"
+        hClose hdl
+        removeFile fp
```

Comment:
What's going on here? Why do we open a file, then close it and remove it?

Reply:
The goal here is to get a path to the temp file. The file itself is created later. There is no simple API for path generation. The file cannot be left empty, because `UserSecret` cannot read empty files.

Reply:
Complicated, can you add a comment about that to this code please?